crates/core/src/rpc/full.rs

```diff
@@ -1642,22 +1642,146 @@ impl Full for SurfpoolFullRpc {
 
     fn get_blocks(
         &self,
-        _meta: Self::Metadata,
-        _start_slot: Slot,
-        _wrapper: Option<RpcBlocksConfigWrapper>,
-        _config: Option<RpcContextConfig>,
+        meta: Self::Metadata,
+        start_slot: Slot,
+        wrapper: Option<RpcBlocksConfigWrapper>,
+        config: Option<RpcContextConfig>,
     ) -> BoxFuture<Result<Vec<Slot>>> {
-        not_implemented_err_async()
+        let svm_locker = match meta.get_svm_locker() {
+            Ok(locker) => locker,
+            Err(e) => return e.into(),
+        };
+
+        Box::pin(async move {
+            let svm_reader = svm_locker.read().await;
+
+            // extract end slot
+            let end_slot = match wrapper {
+                Some(RpcBlocksConfigWrapper::EndSlotOnly(slot)) => slot,
+                Some(RpcBlocksConfigWrapper::ConfigOnly(_)) => None,
+                None => None,
+            };
+
+            let commitment_config = match &wrapper {
+                Some(RpcBlocksConfigWrapper::ConfigOnly(Some(wrapper_config))) => wrapper_config.commitment.clone(),
+                _ => config.and_then(|c| c.commitment),
+            };
+
+            let commitment = commitment_config
+                .unwrap_or_default()
+                .commitment;
+
+            if commitment == solana_commitment_config::CommitmentLevel::Processed {
+                return Err(Error::invalid_params(
+                    "\"processed\" commitment is not supported for getBlocks"
+                ));
+            }
+
+            let current_slot = svm_reader.get_latest_absolute_slot();
+            let slot_threshold = match commitment {
+                solana_commitment_config::CommitmentLevel::Confirmed => current_slot.saturating_sub(1),
+                solana_commitment_config::CommitmentLevel::Finalized => {
+                    current_slot.saturating_sub(FINALIZATION_SLOT_THRESHOLD)
+                }
+                _ => current_slot,
+            };
+
+            // get all available slots from the blocks map
+            let mut available_slots: Vec<Slot> = svm_reader
```

Comment:
I think we'll also need to fetch missing blocks from the remote.

If the user requests blocks 10 - 20, and the lowest block available locally to the surfnet is block 15, we should:
 - get blocks 15 - 20 locally from the surfnet
 - fetch blocks 10-14 from the remote
 - return the combined list